src/meta/attribute.rs

```diff
@@ -1062,23 +1090,159 @@ impl ChannelList {
+    pub fn from_tv50_time(tv50_time_and_flags: u32, user_data: u32) -> Self {
+        Self {
+            drop_frame: false, // do not use bit [6]
+
+            // swap some fields:
+            field_phase: tv50_time_and_flags.get_bit(31),
+            binary_group_flags: [
+                tv50_time_and_flags.get_bit(15),
+                tv50_time_and_flags.get_bit(30),
+                tv50_time_and_flags.get_bit(23),
+            ],
+
+            .. Self::from_tv60_time(tv50_time_and_flags, user_data)
+        }
+    }
+
+
+    /// Pack the SMPTE time code into a u32 value, according to FILM24 packing.
+    /// This encoding does not support the `drop_frame` and `color_frame` flags, they will be lost.
+    /// Panics for invalid values in the time code.
+    pub fn pack_time_as_film24_u32(&self) -> Result<u32> {
+        let mut time = self.pack_time_as_tv60_u32()?;
+        time.set_bit(6, false);
+        time.set_bit(7, false);
+        Ok(time)
+    }
+
+    /// Unpack a time code from one TV60 encoded u32 value and the encoded user data.
+    /// This encoding does not support the `drop_frame` and `color_frame` flags, they will always be `false`.
+    pub fn from_film24_time(film24_time_and_flags: u32, user_data: u32) -> Self {
+        Self {
+            drop_frame: false, // bit [6]
+            color_frame: false, // bit [7]
+            .. Self::from_tv60_time(film24_time_and_flags, user_data)
+        }
+    }
+
+
+    // in rust, group index starts at zero, not at one.
+    fn user_data_bit_indices(group_index: usize) -> std::ops::Range<usize> {
+        let min_bit = 4 * group_index;
+        min_bit .. min_bit + 3
+    }
+
+    /// Pack the user data `u8` array into one u32.
+    /// User data values are clamped to the valid range (maximum value is 4).
+    pub fn pack_user_data_as_u32(&self) -> u32 {
+        let packed = self.binary_groups.iter().enumerate().fold(0_u32, |mut packed, (group_index, group_value)|{
+            packed.set_bits(Self::user_data_bit_indices(group_index), *group_value.min(&4) as u32);
+            packed
+        });
+
+        debug_assert_eq!(Self::unpack_user_data_from_u32(packed), self.binary_groups);
+        packed
+    }
+
+    // Unpack the encoded u32 user data to an array of bytes, each byte having a value from 0 to 4.
+    fn unpack_user_data_from_u32(user_data: u32) -> [u8; 8] {
+        (0..8).map(|group_index| user_data.get_bits(Self::user_data_bit_indices(group_index)) as u8)
+            .collect::<SmallVec<[u8;8]>>().into_inner().expect("array index bug")
+    }
+
+
+    /// Without validation, write this instance to the byte stream, without validation.
     pub fn write<W: Write>(&self, write: &mut W) -> UnitResult {
-        self.time_and_flags.write(write)?;
-        self.user_data.write(write)?;
+        debug_assert!(self.validate(true).is_ok());
+        self.pack_time_as_tv60_u32()?.write(write)?;
```

Comment:
I wrote a long message before realizing the official implementation store timecode as TV60 internally (and so, assume this format), and TV50 and FILM24 are just bit reorganization.

Can I suggest to add a comment like "In official implementation, internal representation of time_and_flags is TV60, this is also how it's written in EXR file".

This is obvious on official implementation because timecode is not stored as multiple fields, but a single int. Here we choose the struct+field approach, so a reminder of the serialized format would make things clear: It's how the official implementation store it.

Reply:
Haha :D yes sounds reasonable, I'll do that